src/main/scala/sigmastate/lang/SigmaTyper.scala

```diff
@@ -23,7 +23,7 @@ class SigmaTyper(val builder: SigmaBuilder) {
 
   private val tT = STypeIdent("T") // to be used in typing rules
 
-  private val predefFuncRegistry = new PredefinedFuncRegistry(builder)
+  private implicit val predefFuncRegistry: PredefinedFuncRegistry = new PredefinedFuncRegistry(builder)
```

Comment:
@greenhat I think it is better to be a parameter of SigmaTyper. Next to SigmaBuilder.

Reply:
@ergomorphic Good idea! Thanks. Made it for binder as well.